packages/core/src/package/preDeployer/PicklistEnabler.ts

```diff
@@ -0,0 +1,156 @@
+import SFPLogger, { Logger, LoggerLevel } from '@dxatscale/sfp-logger';
+import { ComponentSet, registry } from '@salesforce/source-deploy-retrieve';
+import SfpPackage, { PackageType } from '../SfpPackage';
+import { Connection } from '@salesforce/core';
+import { PreDeployer } from './PreDeployer';
+import { Schema } from 'jsforce';
+import QueryHelper from '../../queryHelper/QueryHelper';
+
+const QUERY_BODY =
+    'SELECT Id FROM FieldDefinition WHERE EntityDefinition.QualifiedApiName = ';
+
+
+export default class PicklistEnabler implements PreDeployer {
+    public async isEnabled(sfpPackage: SfpPackage, conn: Connection<Schema>, logger: Logger): Promise<boolean> {
+
+        if (sfpPackage.packageType === PackageType.Unlocked) {
+            if (
+                sfpPackage.isPickListsFound &&
+                (sfpPackage.packageDescriptor.enablePicklist == undefined || sfpPackage.packageDescriptor.enablePicklist == true)
+            ) {
+                return true;
+            }
+        }
+        else
+          return false;
+    }
+
+    public async execute(
+        componentSet: ComponentSet,
+        conn: Connection,
+        logger: Logger
+    ) {
+
+        try {
+            let sourceComponents = componentSet.getSourceComponents().toArray();
+            let components = [];
```

Comment:
@MengAQi  I think this should be optimized, as the during the build phase we already know the components, there is no need to analyze yet again